Refactor some code

lokeshh opened the pull request:
I had a look at some places where code can be improved while documenting the functions. This PR is for that purpose.

lokeshh commented on lib/statsample-glm/glm/base.rb:
```diff
@@ -25,11 +25,9 @@ def initialize ds, y, opts={}
+        @regression = ['Statsample', 'GLM', algorithm, method]
+                      .reduce(Module, :const_get)
+                      .new @data_set, @dependent, @opts
       end
```
@v0dro I see your comment to remove const_get. Is there a way to do it without using const_get?

v0dro replied:
http://stackoverflow.com/a/24420705/3992369

v0dro replied:
Remove the need of chaining basically.

lokeshh replied:
Is the current change fine or should I change to something like `Object.const_get('Daru::Vector')`?

v0dro replied:
Keep a single `const_get`. In other words go with the SO answer.

lokeshh commented:
Changed `degree_of_freedom` to `degrees_of_freedom` and `standard_error` to `standard_errors`. Also added aliases so the old names could be used. I find the tests for these functions are missing.

Should I add the tests? Also should I print a error message to tell that the old names will be removed?

v0dro commented:
Aren't [these](https://github.com/SciRuby/statsample-glm/pull/28/files#diff-a44833ce44ccee8c9a9e6416a838547eR65) the tests you're looking for? Yes print an error message into $stderr.

agisga commented:
I saw too that not every method is tested for every model (logistic, probit, normal etc.), and in general there is a lot of room for improvement of the tests. For example `#standard_errors` is only tested for logistic regression, and I think you are right that `#degrees_of_freedom` is never tested.
Though maybe addition of missing tests should be a separate commit (to make the commit history more organized)?

v0dro commented:
Ah yes there's no tests but those for logit. And new tests should be a separate commit.

lokeshh commented:
I think this is ready to be merged now. Here are the changes:
- Replace multiple `const_get`with single `const_get`.
- Deprecate `degree_of_freedom` and introduce `degrees_of_freedom`
- Deprecate `standard_error` and introduce `standard_errors`
